File: Leetcode/All/001_Two_Sum.java
public class Solution {
    // O(n) time, O(n) space
    public int[] twoSum(int[] nums, int target) {
        int[] ret = new int[2];
        if(nums == null || nums.length == 0)
            return ret;
        
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i++){
            if(map.containsKey(target - nums[i])){
                ret[0] = map.get(target - nums[i]) + 1;
                ret[1] = i + 1;
            }
            map.put(nums[i], i);
        }
        return ret;
    }
}

File: Leetcode/Array/001_Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ret;
        unordered_map<int, int> map;
        for(int i = 0; i < nums.size(); i++){
            if(map.find(target - nums[i]) != map.end()){
                ret.push_back(map[target - nums[i]] + 1);
                ret.push_back(i+1);
                break;
            }
            map[nums[i]] = i;
        }
        return ret;
    }
};